nodes/list_main/statistics.py

```diff
@@ -0,0 +1,194 @@
+    "MEDIAN":             (23, lambda l: get_median(l)),
+    "PERCENTILE":         (24, lambda l, p: get_percentile(l, p))
+}
+
+
+def get_sum(values):
+    return sum(values)
+
+
+def get_sum_of_squares(values):
+    return sum([v * v for v in values])
+
+
+def get_sum_of_inversions(values):
+    return sum([1.0 / v for v in values])
+
+
+def get_product(values):
+    return reduce((lambda x, y: x * y), values)
+
+
+def get_average(values):
+    return sum(values) / len(values)
+
+
+def get_geometric_mean(values):
+    return pow(get_product(values), 1.0 / len(values))
+
+
+def get_harmonic_mean(values):
+    return len(values) / get_sum_of_inversions(values)
+
+
+def get_standard_deviation(values):
+    a = get_average(values)
+    return sqrt(sum([(v - a)**2 for v in values]))
+
+
+def get_root_mean_square(values):
+    return sqrt(get_sum_of_squares(values) / len(values))
+
+
+def get_skewness(values):
+    a = get_average(values)
+    n = len(values)
+    s = get_standard_deviation(values)
+    return sum([(v - a)**3 for v in values]) / n / pow(s, 3)
+
+
+def get_kurtosis(values):
+    a = get_average(values)
+    n = len(values)
+    s = get_standard_deviation(values)
+    return sum([(v - a)**4 for v in values]) / n / pow(s, 4)
+
+
+def get_minimum(values):
+    return min(values)
+
+
+def get_maximum(values):
+    return max(values)
+
+
+def get_median(values):
+    sortedValues = sorted(values)
+    index = int(floor(len(values) / 2))
+    print("index=", index)
+    if len(values) % 2 == 0:  # even number of values ? => take the average of central values
+        median = (sortedValues[index - 1] + sortedValues[index]) / 2
+    else:  # odd number of values ? => take the central value
+        median = sortedValues[index]
+
+    return median
+
+
+def get_percentile(values, percentage):
+    sortedValues = sorted(values)
+    index = int(min(int(floor(len(values) * percentage)), len(values)-1))
+    print("index=", index)
+    return sortedValues[index]
+
+
+class SvStatisticsNode(bpy.types.Node, SverchCustomTreeNode):
+    '''
+    Triggers: Sum, Avg, Min, Max
+    Tooltip: Statistical quantities: sum, average, standard deviation, min, max, product...
+    '''
+    bl_idname = 'SvStatisticsNode'
+    bl_label = 'List Statistics'
+    bl_icon = 'OUTLINER_OB_EMPTY'
+
+    def function_items(self, context):
+        return [(k, k.title(), "", "", s[0]) for k, s in sorted(functions.items(), key=lambda k: k[1][0])]
+
+    def update_function(self, context):
+        if self.function == "PERCENTILE":
+            print("show percentage socket")
+            socket = self.inputs["Percentage"]
+            socket.hide_safe = False
+        else:
+            print("hide percentage socket")
+            socket = self.inputs["Percentage"]
+            socket.hide_safe = True
+
+        updateNode(self, context)
+
+    function = EnumProperty(
+        name="Function", items=function_items, update=update_function)
```

Comment:
Sure.. I wanted to have one dictionary for both the function getter and the function drop down items. I wonder if such  optimization is worth it since the drop down operations is occasional and not in the processing loop to slow down the node. I can implement it to still use one dictionary and precompute this function list so it doesn’t get regenerated every time you drop down. I am trying to avoid having to define two lists one that maps the function name to function, and one that maps function name to a function list item.

Reply:
don't see it as an optimization, see it as consistency.  but it's up to you.

Reply:
well.. i forgot that this also updates whenever the viewport updates. like whenever you move your mouse. So.. 
```
    def function_items(self, context):
        print('i happen more often than you think')
        return [(k, k.title(), "", "", s[0]) for k, s in sorted(functions.items(), key=lambda k: k[1][0])]
```

Reply:
I changed it to use a precomputed list